oci/spec_opts.go

```diff
@@ -1262,15 +1262,15 @@ var ErrNoShmMount = errors.New("no /dev/shm mount specified")
 // The size value is specified in kb, kilobytes.
 func WithDevShmSize(kb int64) SpecOpts {
 	return func(ctx context.Context, _ Client, c *containers.Container, s *Spec) error {
-		for _, m := range s.Mounts {
-			if m.Source == "shm" && m.Type == "tmpfs" {
-				for i, o := range m.Options {
-					if strings.HasPrefix(o, "size=") {
-						m.Options[i] = fmt.Sprintf("size=%dk", kb)
-						return nil
+		for mi, m := range s.Mounts {
+			if m.Destination == "/dev/shm" && m.Source == "shm" && m.Type == "tmpfs" {
+				for oi := 0; oi < len(m.Options); oi++ {
+					if strings.HasPrefix(m.Options[oi], "size=") {
+						m.Options = append(m.Options[:oi], m.Options[oi+1:]...)
```

Comment:
m.Options[i] shares the same memory with m. I don't think we should use append to reallocate memory here.

Reply:
Don't get your point about `should be de-duplicated to prevent overwriting of the latter options`. This WithDevShmSize is to modify the option ...

Reply:
yes，`m.Options` shares the same memory ，but if we want to have the ability to de-duplicate, we need to remove all size options and then add a size option，and the code is also more concise

`append(m.Options[:io], m.Options[oi+1:]...)` does not reallocate memory，only copy

Reply:
OK. Please add unit test for your changes. Thanks